Remotion/Web/Development.WebTesting/ScreenshotCreation/BrowserContentLocators/EdgeBrowserContentLocator.cs

```diff
@@ -0,0 +1,112 @@
+  {
+    private const string c_setWindowTitle =
+        "var w = window; while (w.frameElement) w = w.frameElement.ownerDocument.defaultView; var t = w.document.title; w.document.title = arguments[0]; return t;";
+
+    public EdgeBrowserContentLocator ()
+    {
+    }
+
+    /// <inheritdoc />
+    public Rectangle GetBrowserContentBounds (IWebDriver driver)
+    {
+      var windows = AutomationElement.RootElement.FindAll (
+              TreeScope.Children,
+              new AndCondition (
+                  new OrCondition (
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Window),
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Pane)),
+                  new PropertyCondition (AutomationElement.ClassNameProperty, "Chrome_WidgetWin_1")))
+          .Cast<AutomationElement>()
+          .ToArray();
+
+      if (windows.Length == 1)
+        return ResolveBoundsFromWindow (windows[0]);
+
+      if (windows.Length == 0)
+        throw new InvalidOperationException ("Could not find an Edge browser window in order to resolve the bounds of the content area.");
+
+      // If the result are ambiguous we try to find the browser by changing the window title
+      var automationElement = ResolveByChangingWindowTitle (driver, windows);
+
+      return ResolveBoundsFromWindow (automationElement);
+    }
+
+    [CanBeNull]
+    private AutomationElement ResolveByChangingWindowTitle (IWebDriver driver, IReadOnlyCollection<AutomationElement> windows)
+    {
+      var id = Guid.NewGuid().ToString();
+
+      var executor = (IJavaScriptExecutor) driver;
+      var previousTitle = JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, id);
+
+      var result = RetryUntil (
+          () => windows.SingleOrDefault (w => w.Current.Name.StartsWith (id)),
+          null,
+          3,
+          TimeSpan.FromMilliseconds (100));
+
+      JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, previousTitle);
+
+      if (result == null)
+        throw new InvalidOperationException ("Could not find a matching Edge window by changing its window title.");
+
+      return result;
+    }
+
+    private Rectangle ResolveBoundsFromWindow (AutomationElement window)
+    {
+      var element = RetryUntil (
+          () => window.FindFirst (
+              TreeScope.Subtree,
+              new AndCondition (
+                  new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Document),
+                  new PropertyCondition (AutomationElement.FrameworkIdProperty, "Chrome"))),
+          null,
+          5,
+          TimeSpan.Zero);
+
+      if (element == null)
+        throw new InvalidOperationException ("Could not find the content window of the found Edge browser window.");
+
+      var rawBounds = RetryUntil (() => element.Current.BoundingRectangle, Rect.Empty, 3, TimeSpan.FromMilliseconds (100));

```

Comment:
Is the retry really necessary at this point? Since the object is already there and its `BoundingRectangle` property shouldn't change anymore, unless it is evaluated lazily?

Reply:
It is necessary as the value is evaluated upon accessing it.